spec/recentReleaseSpec.js

```diff
@@ -0,0 +1,184 @@
+    {"id":6732,"name":"Rise & Shine","url":"https://www.igdb.com/games/rise-and-shine","aggregated_rating":63.5714285714286,"release_dates":[{"category":0,"platform":6,"date":1484265600000,"region":8,"human":"2017-Jan-13","y":2017,"m":1},{"category":0,"platform":49,"date":1484265600000,"region":8,"human":"2017-Jan-13","y":2017,"m":1},{"category":2,"platform":3,"date":1514678400000,"human":"2017","y":2017,"m":12},{"category":2,"platform":14,"date":1514678400000,"human":"2017","y":2017,"m":12},{"category":2,"platform":48,"date":1514678400000,"human":"2017","y":2017,"m":12}],"cover":{"cloudinary_id":"m3pcyvvbezj3fbnxsofm"}}
+    ];
+    beforeEach(module('gameJudgement'));
+    beforeEach(inject(function($injector){
+        
+        $httpBackend = $injector.get('$httpBackend');
+        
+        $rootScope = $injector.get('$rootScope');
+        $scope = $rootScope.$new();
+        var $controller = $injector.get('$controller');
+        
+        var steamListUrl = 'steamList.json';
+        
+        requestHandler = { getReleases : function(){
+                $httpBackend.expectGET('api.json',{
+                'X-Mashape-Key': 'MY_IGDB_API_KEY',
+                'Accept' : 'application/json'
+                }).respond(function(status, data, headers, statusText) {
+                    
+                    data = releaseArray;
+                    return [200,{data: releaseArray},'OK'];
+                });
+            },
+            getSteamList : function(){
+                
+                $httpBackend.expectGET(steamListUrl)
+                .respond(function(status, data, headers, statusText) {
+                    
+                    return [200, 'OK'];
+                });
+            }
+        };
+        $scope.newReleases = {data : releaseArray};
+        
+        spyOn(requestHandler,'getReleases');
+        spyOn(requestHandler,'getSteamList');
+        
+        requestHandler.getReleases();
+        requestHandler.getSteamList();
+        
+        createController = function(){
+            return $controller('recentReleasesCtrl',{'$scope' : $scope});
+        };
+        
+    }));
+    
+    it("Should be true if the request for recent releases was sent", function(){
+        
+        expect(requestHandler.getReleases).toHaveBeenCalled();
+    });
+    
+    it("Should be true if the request for the list of steam games was fulfilled", function() {
+        
+        expect(requestHandler.getSteamList).toHaveBeenCalled();
+    });
+    
+    it("Should be true if the press score and the player's score don't exist",function() {
+       
+       expect(!$scope.newReleases.data[8].aggregated_rating && !$scope.newReleases.data[8].rating).toBe(true); 
+    });
+    
+    it("Should be true if a game is on PC but does not exist on steam", function() {
+        
+    });
+    
+    it("Should be true if the press score exists", function() {
```

Comment:
In the beforeEach block of this spec file, I am making the request which returns a status code, the content of my response and OK.